nova_plugin/server.py

```diff
@@ -157,18 +155,22 @@ def _prepare_server_nics(neutron_client, ctx, server):
         ctx, PORT_OPENSTACK_TYPE)
     management_network_id, management_network_name = \
         _get_management_network_id_and_name(neutron_client, ctx)
-    if management_network_id is None and (network_ids or port_ids):
-        # Known limitation
-        raise NonRecoverableError(
-            "Nova server with NICs requires "
-            "'management_network_name' in properties or id "
-            "from provider context, which was not supplied")
+
+    if management_network_id or management_network_name:
+        ctx.logger.debug('Management Network id deprecated and was provided.')
```

Comment:
1. Should be `warning`, not `debug`
2. There was never the ability to specify a management network ID, only a name.

Proposal:

`ctx.logger.warning('management_network_name provided ({0}), however this property is deprecated. Instead, use a relationship to a Network node template'.format(management_network_name))`